### DISCUSSION TITLE
Does the exe need to be packaged or just the binary

### DISCUSSION THREAD

**Discussion by PrimeTheFirst** [aguilaair/updat#23] [Q&A]
Do I need to package the binary generated by flutter build or do I need to use inno setup actions?? Also, opening a discussion would be very helpful.

**Comment by aguilaair** [comment]
I've opened up the discussion section (and I will convert this to a discussion)

**Comment by aguilaair** [comment]
Regarding your question: You need inno (or something similar) to actually install the binary into the system.

Updat's role is simply to fetch the latest version, downalond and run the installer

**Comment by PrimeTheFirst** [comment]
If so, a tutorial would be massively helpful. In any case, I am using github actions to automate the same, and I have managed to create the binaries and have learnt to make a release. Now I need to package the binaries. I have learnt that for windows you can use inno bundle actions, but I dont know how to use it. It doesnt work. And I dont know how to do the same for linux and macos. I only need to do it for those 3. If you have done any such thing during testing, the workflow file or some snippet would be helpful, as I am unable to make any headway. Thanks

**Reply by aguilaair** [reply]
Take a look at: https://github.com/aguilaair/updat/wiki/Github-Actions-Release-Example

It shows how Sidekick packages itself for Windows, Mac and Linux

**Reply by PrimeTheFirst** [reply]
https://github.com/aguilaair/updat/wiki/Github-Actions-Release-Example#:~:text=devbotsxyz/xcode%2Dselect%40v1.1.0
Right here. The action does not exist. I switched it for:
```
      - name: Setup Xcode version
        uses: maxim-lobanov/setup-xcode@v1.6.0
        with:
          xcode-version: latest-stable
```
I see you have used a different one for the actual repo, but forgot to change the wiki.

Also, here: https://github.com/fluttertools/sidekick/blob/356124bea7bc3c9bcd0373f1903a836d2204c569/.github/workflows/release.yml#L138,
 I get
`Run cider version v1.0.16
FormatException: Could not parse "v1.0.16".
Error: Process completed with exit code 70.`

**Reply by aguilaair** [reply]
you can't have the lading v in "v1.0.16"

**Comment by PrimeTheFirst** [comment]
Ah my mistake. I changed that and now it says:
1) Package build_runner is missing(for all three).
 soln. I 'flutter pub add build_runner' before the get packages
 
 2) Macos says /scripts/create_mac_dmg.sh is not found in the create dmg step
 
 3) MSIX is missing
  soln. 'flutter pub add msix' after I added build runner.
  
  
  Errors now:
  macos: ![image](https://github.com/aguilaair/updat/assets/89402723/89d9463b-d354-4e98-92f2-8ea414ee0bd8)
  windows: 
![image](https://github.com/aguilaair/updat/assets/89402723/5b3267e1-431e-4082-a8d1-85852c53e9be)
I have no idea whats going on with linux as the runner is not picking up the job. Also, I changed the one thing that made my code originally different. Im now using boundfox/xcode thingy too.

**Reply by aguilaair** [reply]
you're missing:
Linux:
- https://github.com/fluttertools/sidekick/tree/main/snap
- https://github.com/fluttertools/sidekick/blob/main/AppImageBuilder.yml
- https://github.com/fluttertools/sidekick/blob/main/Sidekick.desktop

Mac:
- https://github.com/fluttertools/sidekick/blob/main/scripts/create_mac_dmg.sh

All:
- https://github.com/fluttertools/sidekick/tree/main/assets (icons, logos, etc.)

**Reply by PrimeTheFirst** [reply]
windows:
![image](https://github.com/aguilaair/updat/assets/89402723/2859cafc-769d-475a-90b9-f46971772b99)


Maos:
![image](https://github.com/aguilaair/updat/assets/89402723/99789c32-f599-49cc-a17c-626acd777b2a)

**Reply by PrimeTheFirst** [reply]
Hey , anything? I know I'm asking a lot of stuff, cause I have no idea what to do next.

**Reply by aguilaair** [reply]
for windows, check your cerificate, there is also an open issue (which sidekick is also facing)

For macos, try using `chmod +x ./scripts/create_mac_dmg.sh && ./scripts/create_mac_dmg.sh`

**Reply by aguilaair** [reply]
It's okay for you to ask! Just understand that I may take some time to reply sometimes :)

**Comment by PrimeTheFirst** [comment]
Hey! This seems like a long process. I'll get to it as soon as I can. Thanks!
